File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.Stack;

class Solution{
    public int solution(String s) {
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            char temp = s.charAt(i);

            if(!stack.isEmpty() && temp == stack.peek()) {
                stack.pop();
            } else {
                stack.push(temp);
            }
        }

        return stack.isEmpty() ? 1 : 0;
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
    const arr = s.split('');
    let stack = [];
    for (let i = 0; i < arr.length; i++){
        const val = arr[i];
        if( val === stack[stack.length - 1]) {
            stack.pop();
        } else {
            stack.push(val);
        }
    }    
    return stack.length === 0 ? 1 : 0 ;
}